### ISSUE TITLE
[FEATURE REQUEST] Changing loglevel during runtime.

### ISSUE THREAD

**Issue by ly4she** [iamolegga/nestjs-pino#371]
Is there any way to dynamically change `loglevel `during runtime? Like in pino using `logger.level`.

https://github.com/pinojs/pino/blob/master/docs/api.md#loggerlevel-string-gettersetter

It will be very useful in my case, when I want to change `loglevel `in working microservice without its reloading. Now I set it with `.env` and there is no way to change it dynamicly.

**Comment by iamolegga** [comment]
No, there is not such feature so far, because there is not direct access to logger instance, more over there are two underlying logger instances: one for http context, one for the rest. And changing level of one of them should change level of another one for expected consistency.

I'll convert this question to feature request and leave it open, if there would be more requests for that, I can comeback to this later. Or you can try implement this on your own, PR are always welcome

**Comment by ly4she** [comment]
@iamolegga thanks for quick response. Ok, I've got you. If start to working on this feature I'll notify you to avoid double work on same feature.

**Comment by jakobwgnr** [comment]
@iamolegga +1 for this, as I think changing log-level at runtime is a feature which a productive app should have. 
I myself am bit unclear on how/which of the several solutions for pino should be incorporated in this package.
BR Jakob
P.S. thanks for creating this package

**Comment by DimosthenisK** [comment]
Now that `logger` is accesible you can do `logger.logger.level = level;`

**Comment by iamolegga** [comment]
> Now that logger is accesible you can do logger.logger.level = level;

good idea, but need to check if this is root `pino` instance and not child logger of it

**Comment by DimosthenisK** [comment]
> > Now that logger is accesible you can do logger.logger.level = level;
> 
> good idea, but need to check if this is root `pino` instance and not child logger of it

Not sure about that, but it definitely works. I just tested it.

**Comment by jakobwgnr** [comment]
@DimosthenisK , @iamolegga I'm using the usecase where Nest Logger is used. https://github.com/iamolegga/nestjs-pino#usage-as-nestjs-app-logger
Doesn't seem to work in this case. Any idea?

**Comment by iamolegga** [comment]
Just released 2.2.0 with this feature. [Docs](https://github.com/iamolegga/nestjs-pino#change-pino-params-at-runtime)

**Comment by StiliyanKushev** [comment]
This doesn't work with `@InjectPinoLogger`.

**Comment by StiliyanKushev** [comment]
This actually still doesn't work well because it:
A) It doesn't handle child loggers (created using `@InjectPinoLogger`)
B) Doesn't allow for dynamic changes due to `AsyncLocalStorage` restrictions ([ref](https://github.com/iamolegga/nestjs-pino/blob/36fd5463ef8ae4cd22e44721f82f33b4e583613a/src/PinoLogger.ts#L85))

Assume we have the following controller:
```ts
@Controller('logger')
export class ApiLoggerController {
	constructor(private readonly apiLoggerService: ApiLoggerService) {}

	// FIXME: use dto
	@Patch('/change-level/:level')
	async setLevel(@Param('level') level: string) {
		this.apiLoggerService.setLevel(level);
	}

	@Get('/level')
	getLevel() {
		return this.apiLoggerService.getLevel();
	}
}
```

And a service that is able to find all child logger instances, cache them, and alter their log level dynamically at runtime.
```ts
import { Injectable, OnApplicationBootstrap } from '@nestjs/common';
import { DiscoveryService } from '@nestjs/core';
import { PinoLogger } from 'nestjs-pino';
import pino from 'pino';

@Injectable()
export class ApiLoggerService implements OnApplicationBootstrap {
	constructor(private readonly discoveryService: DiscoveryService) {}

	private loggerInstances = new Set<PinoLogger>();

	onApplicationBootstrap() {
		const providers = this.discoveryService.getProviders();

		providers.forEach((wrapper) => {
			const instance = wrapper.instance;
			if (!instance) return;

			Object.values(wrapper.instance).forEach((property) => {
				if (property instanceof PinoLogger) {
					this.loggerInstances.add(property);
				}
			});
		});
	}

	setLevel(level: pino.LevelWithSilentOrString) {
		PinoLogger.root.level = level;
		for (const logger of this.loggerInstances) {
			logger.logger.level = level;
		}
	}

	getLevel() {
		return PinoLogger.root.level;
	}
}
```

The function `setLevel()` **only** works if called outside a controller handler, otherwise it will be trapped in the async local storage I mentioned, and will not be able to actually change the logger level.

In other words, the given example above does **NOT** work when you trigger the controller handler method with something like:
> curl -X PATCH localhost:3000/api/logger/change-level/info

Now to be fair, it does change the level but only for the instance(s) in the async local storage, not outside them.
